### PR TITLE
logique du login 

### PR DESCRIPTION
**Pull request by MrPuffs**

### CODE REVIEW

File: lib/widgets/screens/login_view.dart [hunk 1]
```diff
@@ -96,14 +145,17 @@ class _LoginViewState extends State<LoginView> {
...
                             IceButton(
-                                text: connectionButton,
-                                onPressed: () {
-                                  Navigator.pushNamed(
-                                    context,
-                                    '/',
-                                  );
+                                text: _connectionLabelBtn,
+                                onPressed: () async {
+                                  onConnection();
```
Comment (Ch0ufleur) [inline lib/widgets/screens/login_view.dart:158]: il faut await la fonction onConnection car elle est async

**Review by Ch0ufleur — CHANGES_REQUESTED** [review]

File: lib/constants/lang_fr.dart [hunk 2]
```diff
@@ -71,6 +73,13 @@ const String tryLater = 'Veuillez réessayer plus tard.';
...
-const String weakPasswordException = "Le mot de passe ne respecte pas les critères minimales.";
-const String invalidEmailException = "L'email n'est pas valide. Veuillez le changer.";
+const String weakPasswordException =
+    "Le mot de passe ne respecte pas les critères minimales.";
+const String invalidEmailException =
+    "L'email n'est pas valide. Veuillez le changer.";
 const String nullUserException = "L'utilisateur n'a pas été trouvé.";
+const String userNotFoundExeption = "L'adresse courriel est invalide.";
+const String wrongPasswordExeption = "Mot de passe invalide";
+const String emptyFieldExeption = "Champ(s) vide(s)";
```
Comment (cravethedave) [inline lib/constants/lang_fr.dart:83]: "Exception"

File: lib/exceptions/too_many_attempts.dart [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import 'package:figure_skating_jumps/constants/lang_fr.dart';
+import 'package:figure_skating_jumps/exceptions/ice_exception.dart';
+
+class TooManyAttemptsException implements IceException {
+  @override
+  String get devMessage {
+    return "We have blocked all requests from this device due to unusual activity. Try again later.";
```
Comment (cravethedave) [inline lib/exceptions/too_many_attempts.dart:7]: This sounds like a Ui message. A dev message does not ask to try again later. It is only to inform.

File: lib/exceptions/user_not_found_exeption.dart [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import 'package:figure_skating_jumps/constants/lang_fr.dart';
+import 'package:figure_skating_jumps/exceptions/ice_exception.dart';
+
+class UserNotFoundException implements IceException {
+  @override
+  String get devMessage {
+    return "User has not been found.";
```
Comment (cravethedave) [inline lib/exceptions/user_not_found_exeption.dart:7]: User not found est un peu comme null user exception

File: lib/services/user_client.dart [hunk 5]
```diff
@@ -91,6 +94,18 @@ class UserClient {
...
+        case "user-not-found":
+          throw UserNotFoundException();
+        case "wrong-password":
+          throw WrongPasswordException();
+        case "too-many-requests":
+          throw TooManyAttemptsException();
+        default:
+          developer.log(e.toString());
+          rethrow;
+      }
```
Comment (cravethedave) [inline lib/services/user_client.dart:108]: There is a function for this in ExceptionUtils. Make sure it has the cases.

File: lib/widgets/screens/login_view.dart [hunk 6]
```diff
@@ -131,4 +183,18 @@ class _LoginViewState extends State<LoginView> {
...
+  String? _emailValidator(String? value) {
+    if (value == null || value.trim().isEmpty) {
+      return pleaseFillField;
+    }
+    return null;
+  }
+
+  String? _passValidator(String? value) {
+    if (value == null || value.trim().isEmpty) {
+      return pleaseFillField;
```
Comment (cravethedave) [inline lib/widgets/screens/login_view.dart:196]: What is pleaseFillField, ctrlF found nothing

Comment (Ch0ufleur) [reply lib/widgets/screens/login_view.dart:196]: c'est un string dans lang_fr c'est chill

**Review by Ch0ufleur — APPROVED** [review]

**Review by chrichriGeorgie — APPROVED** [review]

**Pull request merged**